### DISCUSSION TITLE
input Pushbutton -  sometimes doesn't respond

### DISCUSSION THREAD

**Discussion by UserBRMFC** [jpcornil-git/HA-mcp23017#15] [General]
Hi,

I have linked a push button (with external pull-up) to an input and an LED to an output, controlled via node red. Very occasionally the input does not respond and the LED is not controlled, how can I fix this?

**Comment by jpcornil-git** [comment]
Hi,

To better understand your use case; I suppose that you are using a Node-Red addon and this component all within homeassistant, e.g. not an independent Node-Red instance communicating with a homeassistant instance thru MQTT ?

When such event happens, are you missing the input event or is the output event not taking place ? (check with the history tab if the input event of the corresponding input/binary_sensor entity is missing)

Assuming input, one possibility is that your event is too short; inputs are scanned every 100ms by default and therefore if your off-on-off pulse is narrower than that it may fall within two sampling points and be missed.

Cheers,
jpc

**Comment by UserBRMFC** [comment]
Hi,

Node-Red Addon in HA and a IO PI plus on my Rpi. Problem is in the input, can a set the sampling rate faster ?

Grtz,
Frank

**Comment by jpcornil-git** [comment]
Yes, you can change that in const.py, see [L18](https://github.com/jpcornil-git/HA-mcp23017/blob/main/custom_components/mcp23017/const.py#L18)
You should also check the waveform of your input (RC time constant may be too large for a short push and input voltage may not reach mcp23017 threshold)
